solidity/test/bridge/Deployment.test.ts

```diff
@@ -0,0 +1,234 @@
+    walletRegistry = await helpers.contracts.getContract("WalletRegistry")
+
+    expect(deployer.address).to.be.not.equal(
+      governance.address,
+      "deployer is the same as governance"
+    )
+  })
+
+  describe("Bridge", () => {
+    it("should set Bridge proxy admin", async () => {
+      // To let a non-proxy-admin read the admin we have to read it directly from
+      // the storage slot, see: https://docs.openzeppelin.com/contracts/4.x/api/proxy#TransparentUpgradeableProxy-admin--
+      expect(
+        ethers.utils.defaultAbiCoder.decode(
+          ["address"],
+          await ethers.provider.getStorageAt(
+            bridge.address,
+            "0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103"
+          )
+        )[0]
+      ).to.be.equal(
+        proxyAdmin.address,
+        "invalid Bridge proxy admin (read from storage slot)"
+      )
+
+      expect(
+        await bridgeProxy.connect(proxyAdmin.address).callStatic.admin()
+      ).to.be.equal(proxyAdmin.address, "invalid Bridge proxy admin")
+    })
+
+    it("should set ProxyAdmin owner", async () => {
+      expect(await proxyAdmin.owner()).to.be.equal(
+        esdm.address,
+        "invalid ProxyAdmin owner"
+      )
+    })
+
+    it("should set Bridge implementation", async () => {
+      // To let a non-proxy-admin read the implementation we have to read it directly from
+      // the storage slot, see: https://docs.openzeppelin.com/contracts/4.x/api/proxy#TransparentUpgradeableProxy-implementation--
+      expect(
+        ethers.utils.defaultAbiCoder.decode(
+          ["address"],
+          await ethers.provider.getStorageAt(
+            bridge.address,
+            "0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc"
+          )
+        )[0],
+        "invalid Bridge implementation (read from storage slot)"
+      ).to.be.equal(bridgeImplementationAddress)
+
+      expect(
+        await bridgeProxy
+          .connect(proxyAdmin.address)
+          .callStatic.implementation(),
+        "invalid Bridge implementation"
+      ).to.be.equal(bridgeImplementationAddress)
+    })
+
+    it("should set Bridge implementation in ProxyAdmin", async () => {
+      expect(
+        await proxyAdmin.getProxyImplementation(bridgeProxy.address),
+        "invalid proxy implementation"
+      ).to.be.equal(bridgeImplementationAddress)
+    })
+
+    it("should set implementation address different than proxy address", async () => {
+      expect(
+        await bridgeProxy
+          .connect(proxyAdmin.address)
+          .callStatic.implementation(),
+        "invalid implementation"
+      ).to.be.not.equal(bridge.address)
+    })
+
+    it("should set Bridge governance", async () => {
+      expect(await bridge.governance()).to.be.equal(
+        bridgeGovernance.address,
+        "invalid Bridge governance"
+      )
+    })
+
+    it("should set BridgeGovernance owner", async () => {
```

Comment:
What do you think about extracting this to a separate `describe`? Same for `WalletRegistry`

Reply:
[1660c63](https://github.com/keep-network/tbtc-v2/pull/359/commits/1660c63a2d7f94342a0e3c99853b5d341b506ff3)